File: sponge-nashorn/examples/script/js/hello_world.js
/**
 * Sponge Knowledge base 
 * Hello world
 */

var HelloWorldTrigger = Java.extend(Trigger, {
    onConfigure: function(self) {
        self.event = "helloEvent";
    },
    onRun : function(self, event) {
        print(event.get("say"));
    }
});

function onStartup() {
    sponge.event("helloEvent").set("say", "Hello World!").send();
}

File: sponge-kotlin/src/test/java/org/openksavi/sponge/kotlin/examples/HelloWorld.kt
package org.openksavi.sponge.kotlin.examples

import org.openksavi.sponge.event.Event
import org.openksavi.sponge.kotlin.KotlinKnowledgeBase
import org.openksavi.sponge.kotlin.KotlinTrigger

class HelloWorld : KotlinKnowledgeBase() {

    class HelloWorld : KotlinTrigger() {
        override fun onConfigure() {
            setEvent("helloEvent")
        }

        override fun onRun(event: Event?) {
            println(event?.get("say"))
        }
    }

    override fun onStartup() {
        eps.event("helloEvent").set("say", "Hello World!").send()
    }
}